README.md

```diff
@@ -1 +1,133 @@
 # Hardware Event Proxy
+
+Hardware Event Proxy handles Redfish hardware events. It contains a main `hw-event-proxy` module written in Go and a `message-parser` module written in Python.
+
+The `message-parser` module is used to parse messages from Redfish Event Message Registry. At startup, it queries the Redfish API and downloads all the Message Registries (if not already included in Sushy library) including custom registries.
+
+Once subscribed, Redfish events can be received by Webhook located in the `hw-event-proxy` module. If the event received does not contain the Message field, `hw-event-proxy` will send a request with Message ID to `message-parser`. Message Parser uses the Message ID to search in the Message Registries and find the Message and Resolution and pass them back to Hhw-event-proxy. `hw-event-proxy` adds them to the event content and converts the event to Cloud Event and sends it out to AMQP channel.  
+
+
+## Running examples locally
+
+The Hardware Event Proxy works with [Cloud Event Proxy](https://github.com/redhat-cne/cloud-event-proxy).
+Run cloud-event-proxy sidecar and consumer example from the cloud-event-proxy repo for testing locally.
+
+### Set environment variables
+```
+export NODE_NAME=mynode
+export HW_PLUGIN=true; export HW_EVENT_PORT=9087; export CONSUMER_TYPE=HW
+export MSG_PARSER_PORT=9097; export MSG_PARSER_TIMEOUT=10
+export REDFISH_USERNAME=<redfish username>; export REDFISH_PASSWORD=<redfish password>; export REDFISH_HOSTADDR=<BMC host/ip address>
+export LOG_LEVEL=debug
+```
+
+### Install and run Apache Qpid Dispach Router
+```
+sudo dnf install qpid-dispatch-router
```

Comment:
we can point to the new installer for QDR deployemnt. We will have difference cases  in that repo readme

Reply:
done